dask/array/fft.py

```diff
@@ -24,21 +24,24 @@
     """
 
 
-def _fft_wrap(fft_func, dtype, out_chunk_fn):
-    def func(a, n=None, axis=-1):
-        if len(a.chunks[axis]) != 1:
-            raise ValueError(chunk_error % (axis, a.chunks[axis]))
+def _fft_wrap(dtype, out_chunk_fn):
```

Comment:
In the past, I have not seen a great way to go from `float*` to `complex*` that preserves precision (e.g. `float32` to `complex64`). So have wound up handling this [myself]( https://github.com/nanshe-org/nanshe/blob/4998fb87f373d951159c49352ca6cdf28c0a8c62/nanshe/imp/registration.py#L186-L192 ). I think from `complex*` to `float*` this should be easier (e.g. take `real` of `0` or similar), but the other way around ends up being kind of ugly.